gitless/core.py

```diff
@@ -516,6 +516,23 @@ def lookup_branch(self, branch_name):
             self.git_remote.name + '/' + branch_name, pygit2.GIT_BRANCH_REMOTE)
     return RemoteBranch(git_branch, self.gl_repo)
 
+  def lookup_branches(self, branch_names):
+    if not stdout(git('ls-remote', '--heads', self.name, branch_names)):
```

Comment:
You need to update the check. Now that you are checking for multiple branches you'll have to verify that all the branches do indeed exist in the remote and decide what to do if some don't.

You could also look at the code and see where this method is called and determine whether it makes sense to have the check at all. I am wondering, if the remote doesn't have a branch given as input, where does this code fail?

Reply:
The code seems to fail in the git.fetch call, and it complains saying 'invalid refspec'. I'll see how to update the check.